lib/identity_cache/configuration_dsl.rb

```diff
@@ -98,9 +98,13 @@ def cache_has_many(association, options = {})
         options[:embed] = :ids unless options.has_key?(:embed)
         deprecate_embed_option(options, false, :ids)
         options[:inverse_name] ||= self.name.underscore.to_sym
-        unless self.reflect_on_association(association)
+
+        unless association_reflection = self.reflect_on_association(association)
           raise AssociationError, "Association named '#{association}' was not found on #{self.class}"
         end
+        if association_reflection.options[:through]
+          raise UnsupportedAssociationError, "caching through associations isn't supported"
+        end
```

Comment:
This code looks super similar to the one in line 145 could be DRYed up with a private method?